golang/internal/grpc/grpc.go

```diff
@@ -34,34 +34,40 @@ type GrpcConnection struct {
 }
 
 type GrpcConnectionParams struct {
-	nodeID  string
-	address string
-	token   string
+	nodeID    string
+	address   string
+	token     string
+	imageDate string
 }
 
 type (
 	DeployFunc     func(context.Context, *dogger.DeploymentLogger, *v1.DeployImageRequest, *v1.VersionData) error
 	WatchFunc      func(context.Context, string) []*common.ContainerStateItem
 	DeleteFunc     func(context.Context, string, string) error
 	SecretListFunc func(context.Context, string, string) ([]string, error)
+	UpdateFunc     func(context.Context, string) error
+	CloseFunc      func(context.Context, agent.CloseReason) error
 )
 
 type WorkerFunctions struct {
 	Deploy     DeployFunc
 	Watch      WatchFunc
 	Delete     DeleteFunc
 	SecretList SecretListFunc
+	Update     UpdateFunc
+	Close      CloseFunc
 }
 
 type contextKey int
 
 const contextConfigKey contextKey = 0
 
-func GrpcTokenToConnectionParams(grpcToken *config.ValidJWT) *GrpcConnectionParams {
+func GrpcTokenToConnectionParams(grpcToken *config.ValidJWT, agentImageDate string) *GrpcConnectionParams {
 	return &GrpcConnectionParams{
-		nodeID:  grpcToken.Subject,
-		address: grpcToken.Issuer,
-		token:   grpcToken.StringifiedToken,
+		nodeID:    grpcToken.Subject,
+		address:   grpcToken.Issuer,
+		token:     grpcToken.StringifiedToken,
+		imageDate: agentImageDate,
```

Comment:
Image hash or the tag? Because if we'd use the tag we could only see "latest" or "stable" and that's not very informative :D

Reply:
The version is the ugly hash, or whatever used while building it. 
Any of these:
- Version
- CommitHash
- BuildTimestamp
If in development these don't matter, where it matters we have can have them.

Reply:
The agent build version is already sent to crux when connecting and we specifically wanted to include something about the image of the agent and since the tag will most likely be "stable" or "latest" the creation date seemed like the best option.